sdk/src/wallet/operations/transaction/high_level/staking/begin.rs

```diff
@@ -0,0 +1,100 @@
+// Copyright 2024 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use serde::{Deserialize, Serialize};
+
+use crate::{
+    client::{api::PreparedTransactionData, secret::SecretManage},
+    types::block::output::{feature::StakingFeature, AccountId, AccountOutputBuilder},
+    wallet::{types::TransactionWithMetadata, TransactionOptions, Wallet},
+};
+
+/// Parameters for beginning a staking period.
+#[derive(Debug, Clone, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct BeginStakingParams {
+    /// The account id which will become a validator.
+    pub account_id: AccountId,
+    /// The amount of tokens to stake.
+    pub staked_amount: u64,
+    /// The fixed cost of the validator, which it receives as part of its Mana rewards.
+    pub fixed_cost: u64,
+    /// The staking period (in epochs). Will default to the staking unbonding period.
+    pub staking_period: Option<u32>,
+}
+
+impl<S: 'static + SecretManage> Wallet<S>
+where
+    crate::wallet::Error: From<S::Error>,
+    crate::client::Error: From<S::Error>,
+{
+    pub async fn begin_staking(
+        &self,
+        params: BeginStakingParams,
+        options: impl Into<Option<TransactionOptions>> + Send,
+    ) -> crate::wallet::Result<TransactionWithMetadata> {
+        let options = options.into();
+        let prepared = self.prepare_begin_staking(params, options.clone()).await?;
+
+        self.sign_and_submit_transaction(prepared, None, options).await
+    }
+
+    /// Prepares the transaction for [Wallet::begin_staking()].
+    pub async fn prepare_begin_staking(
+        &self,
+        params: BeginStakingParams,
+        options: impl Into<Option<TransactionOptions>> + Send,
+    ) -> crate::wallet::Result<PreparedTransactionData> {
+        log::debug!("[TRANSACTION] prepare_begin_staking");
+
+        let (account_id, account_output_data) = self
+            .data()
+            .await
+            .unspent_account_output(&params.account_id)
+            .map(|data| (params.account_id.or_from_output_id(&data.output_id), data.clone()))
```

Comment:
I don't think you need to do that, the passed AccountId can't be null

Reply:
oh, true 😆